src/util/logger/mod.rs

```diff
@@ -226,97 +220,81 @@ where
     }
 }
 
-/// Returns `true` if message was not empty
-fn print_msg_header(mut rd: &mut RecordDecorator, record: &Record) -> io::Result<bool> {
-    rd.start_timestamp()?;
-    write!(rd, "{}", chrono::Local::now().format(TIMESTAMP_FORMAT))?;
+/// Writes log header to decorator. see [log-header-section](https://github.com/tikv/rfcs/blob/master/text/2018-12-19-unified-log-format.md#log-header-section)
+fn write_log_header(decorator: &mut RecordDecorator, record: &Record) -> io::Result<()> {
+    decorator.start_timestamp()?;
+    write!(
+        decorator,
+        "[{}]",
+        chrono::Local::now().format(TIMESTAMP_FORMAT)
+    )?;
 
-    rd.start_whitespace()?;
-    write!(rd, " ")?;
+    decorator.start_whitespace()?;
+    write!(decorator, " ")?;
 
-    rd.start_level()?;
-    write!(rd, "{}", record.level().as_short_str())?;
+    decorator.start_level()?;
+    write!(decorator, "[{}]", record.level().as_short_str())?;
 
-    rd.start_whitespace()?;
-    write!(rd, " ")?;
+    decorator.start_whitespace()?;
+    write!(decorator, " ")?;
 
-    rd.start_msg()?; // There is no `start_line`.
-    write!(
-        rd,
-        "{}:{}",
-        Path::new(record.file())
-            .file_name()
-            .and_then(|path| path.to_str())
-            .unwrap_or("<error>"),
-        record.line()
-    )?;
+    // Writes source file info.
+    decorator.start_msg()?; // There is no `start_file()` or `start_line()`.
+    if let Some(path) = Path::new(record.file())
+        .file_name()
+        .and_then(|path| path.to_str())
+    {
+        write!(decorator, "[{}:{}]", path, record.line())?
```

Comment:
special characters in file name should be filtered away according to RFC

Reply:
Oh, I missed it.😂